app/src/main/java/com/twilio/video/app/ui/room/RoomEvent.kt

```diff
@@ -1,22 +1,38 @@
 package com.twilio.video.app.ui.room
 
+import com.twilio.video.NetworkQualityLevel
+import com.twilio.video.Participant
 import com.twilio.video.Room
+import com.twilio.video.VideoTrack
 import com.twilio.video.app.data.api.AuthServiceError
-import com.twilio.video.app.participant.ParticipantViewState
 
 sealed class RoomEvent {
 
     object Connecting : RoomEvent()
     data class Connected(
-        val remoteParticipants: List<ParticipantViewState>,
+        val participants: List<Participant>,
         val room: Room,
         val roomName: String
     ) : RoomEvent()
     object Disconnected : RoomEvent()
     object ConnectFailure : RoomEvent()
     data class TokenError(val serviceError: AuthServiceError? = null) : RoomEvent()
-    data class ParticipantConnected(val participant: ParticipantViewState) : RoomEvent()
-    data class ParticipantDisconnected(val participant: ParticipantViewState) : RoomEvent()
-    data class NewRemoteVideoTrack(val participant: ParticipantViewState) : RoomEvent()
-    data class DominantSpeakerChanged(val participant: ParticipantViewState) : RoomEvent()
+    data class DominantSpeakerChanged(val newDominantSpeakerSid: String?) : RoomEvent()
```

Comment:
What is this `data class` thing about?

Reply:
It is one of the coolest constructs in Kotlin!! 😄  It is primarily used for classes that just contain data like these classes above. The compiler will generate `hashCode()`,`equals()`, `toString()`, and `copy()`, which you'd have to implement yourself if using a regular class.